Marlin/src/lcd/extui/lib/dgus_creality/creality_touch/PageHandlers.cpp

```diff
@@ -0,0 +1,350 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (c) 2020 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (c) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+#include "../../../../../inc/MarlinConfigPre.h"
+
+#if ENABLED(DGUS_LCD_UI_CREALITY_TOUCH)
+
+#include "../DGUSDisplayDef.h"
+#include "../DGUSDisplay.h"
+#include "../DGUSScreenHandler.h"
+
+#include "../../../../../module/temperature.h"
+#include "../../../../../module/motion.h"
+#include "../../../../../module/planner.h"
+#include "../../../../../feature/pause.h"
+#include "../../../../../feature/runout.h"
+#include "../../../../../module/settings.h"
+
+#include "../../../../marlinui.h"
+#include "../../../ui_api.h"
+
+#include "PageHandlers.h"
+
+// Definitions of page handlers
+
+void MainMenuHandler(DGUS_VP_Variable &var, unsigned short buttonValue) {
+  // Nothing
+}
+
+void ControlMenuHandler(DGUS_VP_Variable &var, unsigned short buttonValue) {
+  switch (var.VP) {
+    case VP_BUTTON_PREPAREENTERKEY:
+      switch (buttonValue) {
+        case 5:       // About
+          // Automatically handled
+          break;
+
+        case 7:       // Reset to factory settings
+          settings.reset();
+          settings.save();
+          break;
+
+        case 9:       // Back button
+          // TODO: should navigate automatically
+          break;
+      }
+      break;
+
+    case VP_BUTTON_ADJUSTENTERKEY:
+      ScreenHandler.HandleLEDToggle();
+      break;
+  }
+}
+
+void LevelingModeHandler(DGUS_VP_Variable &var, unsigned short buttonValue) {
+  switch (var.VP) {
+    case VP_BUTTON_BEDLEVELKEY:
+      switch (buttonValue) {
+        case 1:
+          if (ExtUI::isAxisPositionKnown(ExtUI::axis_t::X) && ExtUI::isAxisPositionKnown(ExtUI::axis_t::Y))
```

Comment:
Although this logic came from the Creality firmware, I've found that for consistency it is best to always home in the middle of the bed to allow the user to adjust the Z-offset: https://github.com/CR6Community/Marlin/commit/2e1670e318d73221a1ce99d47d7b3c9df892f6ba

Reply:
Also, because after homing the Z is raised, it needs an immediate lower to the Z=0 position: https://github.com/CR6Community/Marlin/commit/6a82fce584a20b922bc67437862252d222f31c45

Reply:
When `Z_SAFE_HOMING` is enabled, homing Z moves to the middle of the bed first, so this is ok.